src/detector.c

```diff
@@ -8,6 +8,7 @@
 #include "box.h"
 #include "demo.h"
 #include "option_list.h"
+#include <dirent.h>
```

Comment:
PR is using dirent.h and code that is not cross-compatible, so is unacceptable as is

Reply:
I can add alternatives to dirent

Reply:
> 

can you please tag what part of the code which is causing issue? I will work on it. plus this is a really old PR it needs rebasing as well

Reply:
I rebased it on latest master for you before commenting the issue

Reply:
just rewrite the code in this file avoiding dirent.h inclusion (not available on windows)

Reply:
okay, thanks for pointing out the issues, I'll work on them